src/NimBLEClient.cpp

```diff
@@ -929,41 +945,45 @@ int NimBLEClient::handleGapEvent(struct ble_gap_event* event, void* arg) {
                     break;
             }
 
+            NIMBLE_LOGD(LOG_TAG, "disconnect; reason=%d, %s", rc, NimBLEUtils::returnCodeToString(rc));
+
+            pClient->m_connHandle         = BLE_HS_CONN_HANDLE_NONE;
```

Comment:
This should be moved to be after the onDisconnect() callback.

Reply:
I think so too, still a work in progress.